docs/content/sips/007-deployment-auth.md

```diff
@@ -0,0 +1,158 @@
+  * If the user is logged into that server, use the stored token for auth.  The username and password are not required and _must not be present_.
+  * If the user is not logged into that server, and a username or password is provided, follow the current behaviour.  Both username and password are required.
+  * If the user is not logged into that server, and no username or password is provided, attempt a login using the same behaviour as `spin login`.  If that succeeds, continue to deployment.  If it fails, exit.
+* If the Hippo URL is **not** specified:
+  * If a username or password is specified, error. A URL is required in this case.
+  * If the user is logged in, use the stored URL and token.
+  * If the user is **not** logged in, attempt to log them into a default URL, such as the default address of a local Fermyon platform. If that fails, print usage information and exit.
+
+> QUESTION: If a user is logged into multiple servers, should we:
+> * Default the URL to the _most recent_ login
+> * Provide a way for the user to say which URL is their default
+> * Require the user to provide a URL
+> * Provide an abstraction (e.g. name, config file path) by which the user can switch between servers
+
+> QUESTION: The last section represents a change in behaviour when no flags are provided. Are we happy with this, or should we direct people to `spin login` and not attempt it automatically?
+
+> QUESTION: Is it correct to reject a username/password if the user is logged in to the chosen server, or should we attempt username/password login in this case?
+
+> QUESTION: What happens if the user is logged in to one server and `$HIPPO_URL` is set to another?  What happens if the user is logged in but the `$HIPPO_USERNAME` is set?  (Thinking here of situations where the user is switching between a username/password server and a device-auth one.)  Should we provide a way to "log into" username/password servers, and use that stored state instead of the environment variables?
+
+## Error flows
+
+### Auth failure on deploy
+
+Deployment would fail if the token was incorrect or expired.
+
+> QUESTION: In this case should we simply fail, or should we attempt login?  _Note:_ if we decide to attempt login we _must_ provide a flag to suppress this (for programmatic consumers who have their own login flows).
+
+It would be useful to have an unambiguous way of reporting this for programmatic consumers (e.g. distinguished error code, JSON output option), so they could display login UI or provide a specific prompt.  That may be a bit much to manage in the first drop though.
+
+## `spin login` detailed considerations
+
+`spin login` is a new command and a relatively complex user experience.  This section describes the behaviour in more detail.
+
+### Browser interaction
+
+As part of device auth flow, the user needs to go to the verification URL in their browser and paste in the code.
+
+> QUESTION: Should we attempt to open the browser automatically?
```

Comment:
I think we should. Most CLI with OAuth integration use common tools like `xopen` to open the default web browser when the user is asked to open a browser window.

We could provide an opt-out flag, though I'm not sure how useful that would be to users.